src/main/kotlin/ecommerce/stripe/StripeClient.kt

```diff
@@ -0,0 +1,107 @@
+package ecommerce.stripe
+
+import com.fasterxml.jackson.databind.DeserializationFeature
+import com.fasterxml.jackson.databind.JsonNode
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import ecommerce.dto.PaymentRequest
+import ecommerce.dto.PaymentResponse
+import ecommerce.dto.StripeErrorInfo
+import ecommerce.dto.StripePaymentIntent
+import ecommerce.exception.StripePaymentException
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.http.HttpHeaders
+import org.springframework.http.MediaType
+import org.springframework.stereotype.Component
+import org.springframework.web.client.RestClient
+import org.springframework.web.client.RestClientResponseException
+import kotlin.jvm.java
+
+@Component
+class StripeClient(
+    @Value("\${stripe.secret-key}")
+    private val stripeKey: String,
+) {
+    private val log = LoggerFactory.getLogger(javaClass)
+    private val restClient = RestClient.create()
+
+    private val mapper = jacksonObjectMapper().configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
+
+    fun createCheckoutSession(req: PaymentRequest): PaymentResponse {
+        val body =
+            listOf(
+                "amount=${req.amount}",
+                "currency=${req.currency}",
+                "payment_method=${req.paymentMethod}",
+                "confirm=true",
+                "automatic_payment_methods[enabled]=true",
+                "automatic_payment_methods[allow_redirects]=never",
+            ).joinToString("&")
+
+        return executeStripeRequest(url = "https://api.stripe.com/v1/payment_intents", body = body)
+    }
+
+    fun confirmPayment(intentId: String): PaymentResponse {
+        return executeStripeRequest(
+            url = "https://api.stripe.com/v1/payment_intents/$intentId/confirm",
+            body = "",
+        )
+    }
+
+    private fun executeStripeRequest(
+        url: String,
+        body: String,
+    ): PaymentResponse {
+        try {
+            val resp =
+                restClient.post()
+                    .uri(url)
+                    .header(HttpHeaders.AUTHORIZATION, "Bearer $stripeKey")
+                    .contentType(MediaType.APPLICATION_FORM_URLENCODED)
+                    .body(body)
+                    .retrieve()
+                    .toEntity(String::class.java)
+
+            val paymentIntent = mapper.readValue(resp.body, StripePaymentIntent::class.java)
```

Comment:
Try receiving it directly as a `StripePaymentIntent` object. Use `@JsonProperty` if necessary.